targets/TARGET_STM/trng_api.c

```diff
@@ -63,23 +66,32 @@ void trng_free(trng_t *obj)
     HAL_RNG_DeInit(&obj->handle);
     /* RNG Peripheral clock disable - assume we're the only users of RNG  */
     __HAL_RCC_RNG_CLK_DISABLE();
+
+    users = 0;
 }
 
 int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
 {
-    int ret;
+    int ret = 0;
+    uint8_t random[4];
```

Comment:
I suspect this way most of the compilers would optimise the out the instructions zeroising the array, declaring `random[]` to be volatile might be necessary to prevent it.

Reply:
yet another good point - thanks !
done